src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const dateParts = date.split(fromSeparator);
+  const obj = {};
+  const result = [];
+
+  for (let i = 0; i <= fromFormat.length - 2; i++) {
```

Comment:
The loop condition `i <= fromFormat.length - 2` might be incorrect if the intention is to iterate over all format parts. Consider changing it to `i < fromFormat.length` to ensure all parts are processed.